src/app/components/SetPassword/index.jsx

```diff
@@ -1,68 +1,66 @@
 import React from "react";
-import { Input, Button, Modal, Form } from "antd";
+import { Input, Button, Form } from "antd";
 
-const SetPassword = ({ onOk, visible }) => {
+import passwordSvc from "../../../common/services/password.svc";
+import messagingSvc from "../../../common/services/messaging.svc";
+
+const SetPassword = ({ onOk }) => {
   const [form] = Form.useForm();
 
-  const onDone = (values) => {
+  const onDone = async (values) => {
     if (values.password === values.confirm) {
-      onOk(values.password);
+      await passwordSvc.init(values.password, values.confirm);
```

Comment:
I still struggle with the password management. 
why is there a password Service and then the messaging service still has to send some cache password message?